gtdynamics/utils/ConstraintSpec.h

```diff
@@ -0,0 +1,25 @@
+/* ----------------------------------------------------------------------------
+ * GTDynamics Copyright 2020, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * See LICENSE for the license information
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file  ConstraintSpec.h
+ * @brief Base class for constraint specification
+ * @author: Dan Barladeanu, Frank Dellaert
+ */
+
+#pragma once
+
+namespace gtdynamics {
+
+class ConstraintSpec {
```

Comment:
/** explain this class and mention derived classes you know of */

Reply:
done